File: 2/Liskov.java
import helper.StringNumber;
import helper.ComplexStringNumber;

class Liskov
{
	public static void main(String[] args)
	{
		ComplexStringNumber csn1 = new ComplexStringNumber("-3.14+5.36i");
		ComplexStringNumber csn2 = new ComplexStringNumber("-3+.19i");
		
		System.out.println(csn1.toString());
		System.out.println(csn2.toString());
		System.out.println(csn1.isGreaterThan(csn2));
	}
}

File: 2/Liskov.cpp
#include <iostream>
#include "StringNumber.hpp"
#include "ComplexStringNumber.hpp"

int main()
{
	ComplexStringNumber num1 = ComplexStringNumber("-12.36+1.2i");
	ComplexStringNumber num2 = ComplexStringNumber("+3-11.1i");
	std::cout << num1.toString() << std::endl;
	std::cout << num2.toString() << std::endl;
	
	std::cout << num1.isGreaterThan(num2) << std::endl;
}